### PR TITLE
Decomp a lot of DF950.c

### PR DESCRIPTION
**Pull request by JaThePlayer**
Decomps most of `DF950.c`.

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 9 functions (+0.03%) on `us`.

✅ This PR fixes 34 warnings!

File: src/DF950.c [hunk 1]
```diff
@@ -51,16 +70,152 @@ s32 D_8014F58C[] = { 0x00002043, 0x00002044 };
...
-                     &D_8014F50C, 0x00020000, &D_8014F514, 0x00020000, &D_8014F51C, 0x00020000, &D_8014F524, 0x00020000,
-                     &D_8014F52C, 0x00030000, &D_8014F538, 0x00020000, &D_8014F540, 0x00020000, &D_8014F548, 0x00030000,
-                     &D_8014F554, 0x00020000, &D_8014F55C, 0x00020000, &D_8014F564, 0x00020000, &D_8014F56C, 0x00020000,
-                     &D_8014F574, 0x00020000, &D_8014F57C, 0x00020000, &D_8014F584, 0x00020000, &D_8014F58C, 0x00020000,
-                     &D_8014F594, 0x00020000, &D_8014F59C, 0x00020000,
-                   };
+typedef struct struct_8014F5A4 {
+    /* 0x00 */ s32* unk_00;
+    /* 0x04 */ s16 unk_04;
+    /* 0x06 */ s16 unk_06;
```
Comment (nanaian) [inline src/DF950.c:76]: We can name these;
```suggestion
    /* 0x00 */ s32* arr;
    /* 0x04 */ s16 arrCount;
    /* 0x06 */ s16 pad_06;
```
(@ethteck I'm assuming this is how we're naming padding?)

File: src/DF950.c [hunk 2]
```diff
@@ -91,21 +273,128 @@ s32 func_80149828(void) {
...
+
+    if (sound != NULL) {
+        sound->flags &= -4;
+        if (!keepPlaying) {
+            snd_stop_sound(sound->soundID);
+        }
+    }
+}
+
+// https://decomp.me/scratch/8V5LN
```
Comment (nanaian) [inline src/DF950.c:381]: Is this the non-matching? lol

**Review by nanaian — COMMENTED** [review]
Looks good

File: src/DF950.c (hunk 1 above)
Comment (JaThePlayer) [reply src/DF950.c:76]: unk_06 is not padding, it's used in a function I tried to match (and then failed)

Comment (JaThePlayer) [reply src/DF950.c]: Struct renamed according to Clover's docs

**Review by nanaian — APPROVED** [review]

File: src/DF950.c [hunk 3]
```diff
@@ -76,9 +231,36 @@ void sfx_reset_door_sounds(void) {
...
+// name might be incorrect?
+void sfx_clear_env_sounds(s16 playSounds) {
+    if (!gGameStatusPtr->isBattle) {
+        D_80159AD0 = D_801598A0;
+    } else {
+        D_80159AD0 = D_801599B8;
+    }
+
+    if (playSounds) {
+        s32 i;
```
Comment (ethteck) [inline src/DF950.c:251]: Small nits: could you put the i decl second and have an empty line after it?

File: src/DF950.c [hunk 4]
```diff
@@ -76,9 +231,36 @@ void sfx_reset_door_sounds(void) {
...
+        SoundInstance* sound = D_80159AD0;
+        for (i = 0; i < MAX_SOUND_INSTANCES; sound++) {
+            if (sound->flags & 1) {
+                snd_start_sound_with_shift(sound->soundID, sound->volume, sound->pan, sound->pitchShift);
+            }
 
-INCLUDE_ASM(s32, "DF950", sfx_clear_env_sounds);
+            i++;
```
Comment (ethteck) [inline src/DF950.c:258]: Does it not work to do `i++, sound++` in the loop definition? Also I assume D_80159AD0[i] doesn't work and you actually need to increment the sound temp? Not ideal but not the first time we've seen it

File: src/DF950.c [hunk 5]
```diff
@@ -91,21 +273,164 @@ s32 func_80149828(void) {
     return D_80159AD4;
 }
 
-INCLUDE_ASM(s32, "DF950", sfx_stop_env_sounds);
+void sfx_stop_env_sounds(void) {
+    s32 i;
```
Comment (ethteck) [inline src/DF950.c:277]: Same notes here basically about the one above. You may be able to initialize sound on line 1 like the prev function too

File: src/DF950.c [hunk 6]
```diff
@@ -91,21 +273,164 @@ s32 func_80149828(void) {
...
+
+    if (!gGameStatusPtr->isBattle) {
+        D_80159AD0 = D_801598A0;
+        
+    } else {
+        D_80159AD0 = D_801599B8;
+    }
+
+    sound = D_80159AD0;
+    for (i = 0; i < MAX_SOUND_INSTANCES; i++) {
```
Comment (ethteck) [inline src/DF950.c:288]: I'll stop leaving these loop comments from here on, but basically, see if you can avoid incrementing a sound temp and do just arr[i]. also ,if you do need to increment the sound temp, try doing i++, sound++

**Review by ethteck — CHANGES_REQUESTED** [review]
Sound stuffffff! Join the sound decompers club 😎

File: src/DF950.c (hunk 5 above)
Comment (JaThePlayer) [reply src/DF950.c:277]: sound can't be initialized on line 1 because D_80159AD0 gets assigned to in the first if statement.

Comment (ethteck) [reply src/DF950.c:277]: Oh I'm dumb. It gets incremented in the loop. okay disregard 😁

File: src/DF950.c [hunk 7]
```diff
@@ -91,50 +272,188 @@ s32 func_80149828(void) {
...
 }
 
-INCLUDE_ASM(s32, "DF950", sfx_stop_env_sounds);
+void sfx_stop_env_sounds(void) {
+    SoundInstance* sound;
+    s32 i;
+
+    if (!gGameStatusPtr->isBattle) {
+        D_80159AD0 = D_801598A0;
+
```
Comment (ethteck) [inline src/DF950.c:281]: This empty line can be removed

**Review by ethteck — APPROVED** [review]

**Pull request merged**